### PR TITLE
Qv 2 plugin library - WIP

### PR DESCRIPTION
**Pull request by ndrabins**
* Creates new library "plugin-library"
* implements a updateMaxVoterWeight function given a walletPublicKey, realm, governance can create instructions for updating voter weight. 

TODO: Still need to handle the lookup and combining of transactions for each plugin + multiple

### CODE REVIEW

File: plugin-library/updateVoterWeightRecord.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    options
+  )
+  console.log('provider', provider)
+
+  // TODO fetch this from the realm governance and perform step checking on each plugin
+  const pluginName = 'gateway'
+  // Get the plugin client
+  const client = await loadClient(pluginName, provider)
+
+  // check if the voter weight record exists already. If not, create it.
```
Comment (dankelleher) [inline plugin-library/updateVoterWeightRecord.tsx:59]: comment in the wrong place I think

File: plugin-library/getPlugins.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  // Find the gatekeeper network from the registrar
+  return registrarObject.previousVoterWeightPluginProgramId
+}
+
+export const getPlugins = async (
+  realmPublicKey: PublicKey,
+  governanceMintPublicKey: PublicKey, // TODO, if this is the council mint, get plugins for the council governance instead of community
+  connection: Connection
+): Promise<string[]> => {
```
Comment (dankelleher) [inline plugin-library/getPlugins.ts:36]: why is this string? shoud it not be PluginName?

File: plugin-library/getPlugins.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    do {
+      pluginName = findPluginName(programId)
+      if (pluginName) {
+        plugins.push(pluginName as PluginName)
+      }
+
+      const client = await loadClient(pluginName as PluginName, provider)
+      // @ts-ignore
+      programId = await getPredecessorProgramId(
```
Comment (dankelleher) [inline plugin-library/getPlugins.ts:62]: can you add a comment on why ts-ignore is needed?

Comment (ndrabins) [reply plugin-library/getPlugins.ts:62]: im going to fix this, the type it expects it something like string | undefined, and it comes back as string | null

Just wanted to make sure I was on the right track before bogged down by typescript issues

**Pull request merged**